src/URIs.jl

```diff
@@ -457,7 +455,30 @@ function absuri(uri::URI, context::URI)
     @assert !isempty(context.host)
     @assert isempty(uri.port)
 
-    return merge(context; path=uri.path, query=uri.query)
+    return URI(context; path=uri.path, query=uri.query)
+end
+
+"""
+    joinpath(uri, path) -> URI
+
+Join the path component of URI and other parts.
+"""
+function Base.joinpath(uri::URI, parts::String...)
+    path = uri.path
+    for p in parts
+        if startswith(p, '/')
+            path = p
+        elseif isempty(path) || path[end] == '/'
+            path *= p
+        else
+            path *= "/" * p
+        end
+    end
+
+    if isempty(uri.path)
+        path = "/" * path
+    end
+    return URI(uri; path=normpath(path))
```

Comment:
I'd rather avoid `normpath` as that uses system-dependent regexes to parse the path (including drive letters and such, on windows).

Reply:
Let's follow the exact rules in RFC1808 section 4 I guess?

https://tools.ietf.org/html/rfc1808#section-4

Reply:
I think this `normpath` calls the `URLs.normpath`: https://github.com/JuliaWeb/URIs.jl/blob/master/src/URIs.jl#L415 which follows https://tools.ietf.org/html/rfc3986#section-5.2.4 already @c42f

Reply:
Oh dear, yes indeed!

This is what happens when trying to maintain a package I seem to have inherited by accident :laughing: